hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/AbstractS3ACommitter.java

```diff
@@ -147,6 +173,11 @@ protected AbstractS3ACommitter(
     this.jobContext = context;
     this.role = "Task committer " + context.getTaskAttemptID();
     setConf(context.getConfiguration());
+    Pair<String, JobUUIDSource> id = buildJobUUID(
+        conf, context.getJobID());
+    uuid = id.getLeft();
+    uuidSource = id.getRight();
```

Comment:
Other places use `this.` as a prefix when setting fields. I find that helpful when reading to know that an instance field is being set, vs a local variable.

Reply:
Makes sense in the constructor. Done